### PR TITLE
Printables context around vouchers

### PR DESCRIPTION
**Pull request by aghimpu**
https://github.com/teamforus/forus/issues/865

### CODE REVIEW

File: src/sources/forus-platform/js/angular-1/i18n/nl/pages/voucher-printable.pug.i18n.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            show_to: 'Laat deze voucher eenmalig zien bij de destreffende aanbieder:',
+            contact_us: 'Neem dan contact op met de aanbieder.'
+        }
+    },
+    budget_voucher: {
+        labels: {
+            use_for: 'Gebruik deze budget voucher voor:',
+            description: 'Dit is uw {{printableTitle}} voucher met een QR-code',
+            show_to: 'Laat deze voucher zien bij een van de destreffende aanbieders, te vinden op:',
+            contact_us: 'Neem dan contact op.'
```
Comment (maxvisser) [inline src/sources/forus-platform/js/angular-1/i18n/nl/pages/voucher-printable.pug.i18n.js]: Neem contact op met:

File: src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug [hunk 2]
```diff
@@ -1,10 +1,40 @@
...
+    
+        qr-code(qr-value="$ctrl.qrCodeValue" qr-type="voucher")
+        .printable-title {{ $ctrl.printableTitle }}
+        .printable-description(
+            i18n="voucher_printable.{{$ctrl.voucherType}}_voucher.labels.description" 
+            i18n-vars="{ printableTitle: $ctrl.printableTitle }")
+
+    .printable-additional-info
+        .description(i18n="voucher_printable.{{$ctrl.voucherType}}_voucher.labels.show_to")  
+        .reference {{ $ctrl.voucherType != 'product' ? $ctrl.organization.site : $ctrl.organization.name }}
```
Comment (maxvisser) [inline src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug]: $ctrl.organization.site is empty on pdf and not displayed 
image

File: src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug [hunk 3]
```diff
@@ -1,10 +1,40 @@
...
-    .printable-title(ng-if="$ctrl.voucher.type != 'product'") Uw {{ $ctrl.voucher.fund.name }}-voucher
-    .printable-title(ng-if="$ctrl.voucher.type == 'product'") Uw {{ $ctrl.voucher.product.name }}-voucher
-    br
-    qr-code(qr-value="$ctrl.qrCodeValue" qr-type="voucher")
-    .printable-description De verloopdatum van deze voucher is {{ $ctrl.voucher.expire_at_locale }}
+    .printable-main-info
+        .printable-usage-info 
+            span(i18n="voucher_printable.{{$ctrl.voucherType}}_voucher.labels.use_for")
+            br
+            strong {{ $ctrl.voucher.expire_at_locale }}
```
Comment (maxvisser) [inline src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug:7]: expire_at is missing font-family.
image

**Review by maxvisser — CHANGES_REQUESTED** [review]

**Comment by maxvisser** [comment]
Could you implement the same on forus-webshop? @aghimpu

**Comment by maxvisser** [comment]
Implementation logo is missing in right corner
image

**Comment by aghimpu** [comment]
> Implementation logo is missing in right corner
> image

Actually it's using the implementation logo, we just don't have them on the platform, I added a transparent image called implementation-logo.png on platform_common. If you need them for all the platforms you can just add the images in the corresponding folders for each platform and it will override the common one.

**Comment by maxvisser** [comment]
> > Implementation logo is missing in right corner
> > image
> 
> Actually it's using the implementation logo, we just don't have them on the platform, I added a transparent image called implementation-logo.png on platform_common. If you need them for all the platforms you can just add the images in the corresponding folders for each platform and it will override the common one.

I think this is a good idea; I just checked none of the implementation logo's are black and white but they are there: (example)
image

File: src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug (hunk 2 above)
Comment (aghimpu) [reply src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug]: Should that block be hidden (for budget vouchers) if the organization doesnt have a website?

**Comment by aghimpu** [comment]
> Could you implement the same on forus-webshop? @aghimpu

Done.

File: src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug (hunk 2 above)
Comment (maxvisser) [reply src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug]: Ah you don't understand which website should be displayed. Its the implementations webshop.
The text above is the dutch translation for "Show this voucher to providers that are approved, you can find this list on:"

Comment (maxvisser) [reply src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug]: It should display the webshop_url in implementation table (or general url if no implementation applicable) image

File: src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug [hunk 4]
```diff
@@ -1,10 +1,40 @@
...
+            br
+            span(i18n="voucher_printable.{{$ctrl.voucherType}}_voucher.labels.contact_us") 
+
+        span {{ "voucher_printable.default.labels.contact.phone" | translate }}: 
+        strong {{ $ctrl.organization.phone }}  
+        br
+        span {{ "voucher_printable.default.labels.contact.email" | translate }}: 
+        strong {{ $ctrl.organization.email }}   
+
+        .general-logo
```
Comment (maxvisser) [inline src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug:36]: _webshop-common\assets\img\me\logo-bw.svg is a forus logo with less contrast; or download the one from figma

File: src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug [hunk 5]
```diff
@@ -1,10 +1,40 @@
...
+    
+        qr-code(qr-value="$ctrl.qrCodeValue" qr-type="voucher")
+        .printable-title {{ $ctrl.printableTitle }}
+        .printable-description(
+            i18n="voucher_printable.{{$ctrl.voucherType}}_voucher.labels.description" 
+            i18n-vars="{ printableTitle: $ctrl.printableTitle }")
+
+    .printable-additional-info(ng-if="$ctrl.voucherType == 'product' || $ctrl.organization.website")
+        .description(i18n="voucher_printable.{{$ctrl.voucherType}}_voucher.labels.show_to")  
+        .reference {{ $ctrl.voucherType != 'product' ? $ctrl.organization.website : $ctrl.organization.name }}
```
Comment (maxvisser) [inline src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug]: should be implementation webshop instead of organization website

**Review by maxvisser — CHANGES_REQUESTED** [review]

**Comment by maxvisser** [comment]
added some bw implementation logos myself

File: src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug (hunk 5 above)
Comment (maxvisser) [reply src/sources/forus-platform/pug/tpl/printables/printable-voucher-qr_code.pug]: https://github.com/teamforus/forus-frontend/pull/817#discussion_r353127085

File: src/sources/forus-platform/js/angular-1/i18n/nl/pages/voucher-printable.pug.i18n.js (hunk 1 above)
Comment (aghimpu) [reply src/sources/forus-platform/js/angular-1/i18n/nl/pages/voucher-printable.pug.i18n.js]: I think that translation was updated on my last commit:

https://github.com/teamforus/forus-frontend/pull/817/commits/232664a03a9509693f0252f040952f7a090b5ccc

https://github.com/teamforus/forus-frontend/blob/fe7d59424cdf9ea848cc3788fab8855b2859a282/src/sources/forus-platform/js/angular-1/i18n/nl/pages/voucher-printable.pug.i18n.js

**Review by maxvisser — APPROVED** [review]

**Pull request merged**